mrp_repair_order/models/repair.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Antoni Marroig(APSL-Nagarro)<amarroig@apsl.net>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class RepairOrder(models.Model):
+    _inherit = "repair.order"
+
+    mrp_count = fields.Integer(
+        "Mrp production order count", compute="_compute_mrp_production_count"
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        self.env["mrp.production"].browse(
+            self.env.context["default_mrp_id"]
+        ).repair_id = res.id
+        return res
+
+    def _compute_mrp_production_count(self):
+        self.mrp_count = len(
+            self.env["mrp.production"].search([("repair_id", "=", self.id)])
```

Comment:
Compute methods expect a record set. You need to loop through self.
I notice you are using a lot of search for the is of the repair order on the manufacturer order. Maybe would be a good idea to create a one2many for production orders and match on the field repair_id . Let Odoo have the records available to you

Reply:
Thanks for the anotations, i make a one2many for production orders.
Can you review again please